### PR TITLE
[Merged by Bors] - Deprecate usage of `svm/state` and replace it with `svm` at calling sites 

### PR DESCRIPTION
**Pull request by neysofu**
## Motivation
We're hard deprecating the `svm/state` package, which will remain in the `go-spacemesh` as a temporary solution before we make the switch to `go-svm`. This task is tracked in https://github.com/spacemeshos/go-spacemesh/issues/2828.

## Changes
Making sure that `svm/state` is not used anywhere in the codebase and that all calling sites interact directly with the `svm` package instead.

## Test Plan
No functional changes are expected. Some tests, however, require changes in how they interact with `svm` as well.

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by neysofu** [comment]
bors try

File: cmd/node/node.go [hunk 1]
```diff
@@ -661,24 +657,23 @@ func (app *App) initServices(ctx context.Context,
...
+	app.host.Register(svm.IncomingTxProtocol, pubsub.ChainGossipHandler(syncHandler, state.HandleGossipTransaction))
 	app.host.Register(activation.PoetProofProtocol, poetListener.HandlePoetProofMessage)
 
 	app.proposalBuilder = proposalBuilder
 	app.blockListener = blockListener
 	app.mesh = msh
 	app.syncer = newSyncer
 	app.clock = clock
-	app.state = processor
+	app.svm = state
```
Comment (noamnelke) [inline cmd/node/node.go:674]: This happens twice - see [line 676](https://github.com/spacemeshos/go-spacemesh/pull/2992/files#diff-69bfff7b16ff2da1ebcca3ff8dcd296aa906d17d4af33acac1b959329fcd95b0R676) below

**Review by noamnelke — APPROVED** [review]
Only one minor comment

File: cmd/node/node.go (hunk 1 above)
Comment (neysofu) [reply cmd/node/node.go:674]: Thanks! Fixing this.

**Comment by neysofu** [comment]
bors merge

**Comment by neysofu** [comment]
bors try

**Comment by neysofu** [comment]
bors merge

**Comment by neysofu** [comment]
bors merge

**Comment by neysofu** [comment]
bors merge

**Comment by neysofu** [comment]
bors try

**Comment by neysofu** [comment]
bors merge

**Pull request closed without merging**